service/dap/server_test.go

```diff
@@ -3335,6 +3334,9 @@ func TestFatalThrowBreakpoint(t *testing.T) {
 					if se.Body.Reason != "exception" || se.Body.Description != "Paused on fatal error" {
 						t.Errorf("\ngot  %#v\nwant Reason=\"exception\" Description=\"Paused on fatal error\"", se)
 					}
+
+					// TODO(suzmue): Get the exception info for the thread and check the description
+					// includes "all goroutines are asleep - deadlock!".
```

Comment:
You should be able to test this, just gate the assertion on the Go compiler being version 1.14 or 1.15 (which we still test via CI).

Reply:
The actual issue that is breaking this test is that the deadlock is not associated with a goroutine ID, so we can't request the exception info. This needs to be fixed too, but makes sense to do separately because we will likely want to add the thread that is running to display the stack trace etc.

Reply:
What about _fixtures/fatalerror.go? Does that one result in a selected goroutine, so you have an id linked to a description that you can check?

Reply:
I don't see that fixture. I could add a fixture to do a concurrent map read / write and I could test it that way.

Reply:
Huh! That's because it was in my local client. I was going to add it at some point, but never did.  This is easier than concurrent map operations:
```
func main() {
    var f func()
    go f()
}
```

Reply:
Added the fixture and the test

Reply:
The exception info is shown in the source code, so if its not associated with a thread with a stack trace, I think there would need to be a separate UI for it.